docs/docs/api/producer.md

```diff
@@ -153,7 +154,13 @@ const producer: Producer = ({
     return;
```

Comment:
I guess this should go on the error path. So before return do an `updateError('Invalid state')` or some sorts.

The idea is that the guard protects the producer from accidental triggering - so when the guard condition is met then the producer has a green light to execute.

However, if, like in the example above, data is invalid then the execution part is faulty. As such the producer should inform if the error is irremediable or have a strategy for mitigating this scenario.

The problem here is that you might have situations where a process halts unexpectedly. Ideally, both valid and non-valid paths should be explicit in the state and in producers so that every process is recoverable or at least inform that the process has failed.